sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -329,6 +329,14 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       Row(3, "c") ::
       Row(4, "d") :: Nil)
     checkAnswer(lowerCaseData.intersect(upperCaseData), Nil)
+
+    // check null equality
```

Comment:
Thank you! Let me add the test case. It passes but we should still add this case!